dist/type_to_str.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+
+import re
+
+"""
+* docstring: The documentation to attach to the generated function.
+* name: The name to use for the generated function, i.e. `__wt_name_str`.
+* define_regex: The regular expression to use for capturing the names of the
+  macros. Note that this must also include the capture group.
+* srcfile: The source file to look through when searching for `define_regex`.
+* out: The output file handle.
+"""
+def generate_string_conversion(docstring, name, define_regex, srcfile, out):
+    srcfile = '../' + srcfile
+    pattern = re.compile(define_regex)
+
+    out.write('/*\n')
+    out.write(' * __wt_{}_str --\n'.format(name))
+    out.write(' *     {}\n'.format(docstring))
+    out.write(' */\n')
+    out.write('static inline const char *\n')
+    out.write('__wt_{}_str(uint8_t val)\n'.format(name))
+    out.write('{\n')
+    out.write('    switch (val) {\n')
+
+    found = False
+    with open(srcfile, 'r') as src:
+        for line in src:
+            matches = pattern.search(line)
+            if matches:
+                found = True
+                out.write('    case {}:\n'.format(matches.group(1)))
+                out.write('        return ("{}");\n'.format(matches.group(1)))
+
+    if not found:
+        raise Exception("generate_string_conversion: Failed to find {} in {}"
+                        .format(define_regex, srcfile))
+
+    out.write('    }\n')
+    out.write('\n')
+    out.write('    return ("{}_INVALID");\n'.format(name.upper()))
+    out.write('}\n')
+
+if __name__ == '__main__':
+    with open('../src/include/str_inline.h', 'w+') as out:
+        generate_string_conversion('Convert a prepare state to its string representation.',
+                                   'prepare_state',
+                                   r'^#define\s+(WT_PREPARE_[A-Z0-9_]+)\s+[0-9]+',
+                                   'src/include/btmem.h',
+                                   out)
+        out.write('\n')
```

Comment:
Other functions that auto-update source files generally create the new version into a temporary file and only swap it into place if it has changed.
That avoids source control (Git) getting surprised by different timestamps, or spurious line ending differences. See dist/api_config.py for an example.

It might no longer be necessary? Maybe Git is smart enough now not to worry?

Reply:
That's fair, I'll do it like `api_config.py`. Git doesn't care about timestamps and the line-ending thing is OK since I'm manually inserting them here, but your overall point still stands.

Reply:
Fixed in a660e6f